### ISSUE TITLE
Misuse of scikit.learn.cluster: graph adjacency matrix is not symmetric

### ISSUE THREAD

**Issue by fritzo** [priorknowledge/loom#92]
<pre>
$ make test
...
~/.virtualenvs/sf/local/lib/python2.7/site-packages/sklearn/manifold/spectral_embedding_.py:220: UserWarning: Graph adjacency matrix should be symmetric. Converted to be symmetric by average with its transpose.
  warnings.warn("Graph adjacency matrix should be symmetric. "
</pre>

**Comment by jglidden** [comment]
Right, the similarity matrix is only approximately symmetric due to Monte Carlo error. Scikit-learn takes the average of the matrix with its transpose which seemed reasonable to me. I agree it's probably better to handle creating a symmetric affinity matrix ourselves explicitly though.
